templates/webapi/comments/pagination.html.ep

```diff
@@ -1,54 +1,5 @@
-<nav class="comments-pagination" aria-label="Comments pagination">
-  <ul class="pagination">
-    % if ($comments_pager->current_page == $comments_pager->first_page) {
-        <li class="disabled"><span
-    % }
-    % else {
-        <li><a href="<%= url_with->query([ comments_page => $comments_pager->current_page - 1 ]) %>"
-    % }
-      class="page-link" aria-label="Previous">
-        <span aria-hidden="true">&laquo;</span>
-        <span class="sr-only">Previous</span>
-    % if ($comments_pager->current_page == $comments_pager->first_page) {
-        </span>
-    % }
-    % else {
-        </a>
-    % }
-
-    % my $skipped = 0;
-    % my $skipping_range = 5;
-    % for (my $i = $comments_pager->first_page; $i <= $comments_pager->last_page; $i++) {
-        % # skip rendering page buttons which are too far from the beginning, the end or the current page
-        % unless ($i < $skipping_range || $i + $skipping_range >= $comments_pager->last_page || (abs($comments_pager->current_page - $i) < $skipping_range)) {
-            % # render … between gaps
-            % unless ($skipped) {
-                <li class="disabled"><span class="page-link">…</span></li>
-                % $skipped = 1;
-            % }
-            % next;
-        % }
-        % $skipped = 0;
-        %= tag 'li', class => ($i == $comments_pager->current_page ? 'active' : undef) => sub {
-        %    link_to $i => url_with->query([ comments_page => $i ]) => (class => 'page-link')
-        % };
-    % }
-
-    % if ($comments_pager->current_page == $comments_pager->last_page) {
-        <li class="disabled"><span
-    % }
-    % else {
-        <li><a href="<%= url_with->query([ comments_page => $comments_pager->current_page + 1 ]) %>"
-    % }
-        class="page-link" aria-label="Next">
-          <span aria-hidden="true">&raquo;</span>
-          <span class="sr-only">Next</span>
-    % if ($comments_pager->current_page == $comments_pager->last_page) {
-        </span>
-    % }
-    % else {
-        </a>
-    % }
-    </li>
-  </ul>
-</nav>
+%= include 'comments/build_pagination', comments_pager => $comments_pager, media_class => "d-block d-sm-none", max => 5;
+%= include 'comments/build_pagination', comments_pager => $comments_pager, media_class => "d-none d-sm-block d-md-none", max => 10;
+%= include 'comments/build_pagination', comments_pager => $comments_pager, media_class => "d-none d-md-block d-lg-none", max => 12;
+%= include 'comments/build_pagination', comments_pager => $comments_pager, media_class => "d-none d-lg-block d-xl-none", max => 16;
+%= include 'comments/build_pagination', comments_pager => $comments_pager, media_class => "d-none d-xl-block", max => 20;
```

Comment:
This looks to render the pagination template 5 times in total?

Reply:
Yes, but each one has it's own media class, so only one will be visible at the same time.
This is the first approach. Now I want to explore how to render it once, and use the css to show only certain numbers of the pagination depending of the media size

Reply:
I would not add multiple versions of the pagination with fixed "media query classes". It is likely better to write the media-queries for the pagination ourselves. We likely don't need anything super fancy here.